src/coreclr/jit/lowerxarch.cpp

```diff
@@ -1631,6 +1779,27 @@ GenTree* Lowering::LowerHWIntrinsic(GenTreeHWIntrinsic* node)
             LowerFusedMultiplyAdd(node);
             break;
 
+        case NI_AVX_BroadcastScalarToVector256:
+        case NI_AVX_BroadcastScalarToVector128:
+        {
+            // there can be 2 cases hitting AVX_BroadcastScalarToVector*
+            // 1. pass the address as LCL_ADDR to AVX.BroadcastScalarToVector256() API
+            // 2. pass the address as LCL_VAR  to AVX.BroadcastScalarToVector256() API
+            LIR::Use use;
+            bool     foundUse = BlockRange().TryGetUse(node, &use);
+            if (foundUse && use.User()->OperIs(GT_HWINTRINSIC) &&
+                use.User()->AsHWIntrinsic()->OperIsEmbBroadcastCompatible())
+            {
+                GenTree* op = node->Op(1);
+                assert(op->OperIs(GT_LCL_ADDR, GT_LCL_VAR));
+                if (node == use.User()->AsHWIntrinsic()->Op(1))
+                {
+                    std::swap(use.User()->AsHWIntrinsic()->Op(1), use.User()->AsHWIntrinsic()->Op(2));
```

Comment:
Same as above, if this is going to be contained, we can't do this "safely" without doing the relevant `IsSafeToContain` checks.

It would be better to have this handled as part of the regular containment logic instead